lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -307,12 +312,36 @@ def from_string(cls, collection_input, artifacts_manager, supplemental_signature
 
     @classmethod
     def from_requirement_dict(cls, collection_req, art_mgr, validate_signature_options=True):
-        req_name = collection_req.get('name', None)
-        req_version = collection_req.get('version', '*')
-        req_type = collection_req.get('type')
+        result = ArgumentSpecValidator(
+            argument_spec=dict(
+                name=dict(type='str', required=True),
```

Comment:
Currently, seeing as `collection_req` is always passed in as a dictionary, I have set up the `argument_spec` to always require a `name` key. I could toggle this if `collection_req` was instead a list of collection names, but I'm not sure where that would get used?